ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/build_run_synthea.sh

```diff
@@ -0,0 +1,472 @@
+do_load_validation(){
+  cd "${BFD_SYNTHEA_AUTO_LOCATION}"
+  # restore the copy of the RIF (.csv) files back to the synthea output directory
+  cp "${BFD_SYNTHEA_OUTPUT_LOCATION}/backup/*.csv" "${BFD_SYNTHEA_OUTPUT_LOCATION}"
+
+  # now perform the validation of the run.
+  END_BENE_ID=$(cat "${BFD_SYNTHEA_OUTPUT_LOCATION}/${BFD_END_STATE_PROPERTIES}" |grep bene_id_start |sed 's/.*=//')
+  echo "END_BENE_ID=${END_BENE_ID}"
+
+  source .venv/bin/activate
+  python3 validate-synthea-load.py "${BEG_BENE_ID}" "${END_BENE_ID}"  "${TARGET_SYNTHEA_DIR}" "${UNIQUE_ENVS_PARAM}"
+  deactivate
+}
+
+# Args:
+# 1: bene id start (inclusive, taken from previous end state properties / synthea properties file)
+# 2: bene id end (exclusive, taken from new output end state properties)
+# 3: file system location to write the characteristics file
+# 4: which environment to check, should be a single value from the list of [test prd-sbx prod]
+#
+# script will check the number of lines written to the characteristics.csv file; if only the header row (row #1)
+# then we'll exit out.
+gen_characteristics_file(){
+  cd "${BFD_SYNTHEA_AUTO_LOCATION}"
+  source .venv/bin/activate
+  python3 generate-characteristics-file.py "${BEG_BENE_ID}" "${END_BENE_ID}"  "${BFD_SYNTHEA_AUTO_LOCATION}" "${UNIQUE_ENVS_PARAM}"
+
+  # check the generated output file; must have more than just the header line
+  line_cnt=$(cat "${BFD_SYNTHEA_AUTO_LOCATION}/${BFD_CHARACTERISTICS_FILE_NAME}" |wc -l)
+  if [[ "$line_cnt" -gt 1 ]]; then
+    BFD_CHARACTERISTICS="${BFD_SYNTHEA_AUTO_LOCATION}/${BFD_CHARACTERISTICS_FILE_NAME}"
+    echo "${BFD_SYNTHEA_OUTPUT_LOCATION}/${BFD_CHARACTERISTICS_FILE_NAME} successfully discovered"
+  else
+    echo "failed to read meaningful data from: ${BFD_SYNTHEA_OUTPUT_LOCATION}/${BFD_CHARACTERISTICS_FILE_NAME}"
+  fi
+  deactivate
+}
+
+# Function to upload the new end state properties file to an S3 bucket; it also extracts
+# the newest bene_id_start variable for use in a later function/operation.
+upload_characteristics_file_to_s3(){
+  echo "upload ${BFD_CHARACTERISTICS_FILE_NAME} file to S3"
+  cd "${BFD_SYNTHEA_AUTO_LOCATION}"
+  source .venv/bin/activate
+  python3 ./s3_utilities.py "${BFD_SYNTHEA_AUTO_LOCATION}" "upload_characteristics"
+  deactivate
+}
+
+# Function to upload the new end state properties file to an S3 bucket; it also extracts
+# the newest bene_id_start variable for use in a later function/operation.
+upload_props_file_to_s3(){
+  echo "upload end_state.properties file to S3"
+  cd "${BFD_SYNTHEA_AUTO_LOCATION}"
+  source .venv/bin/activate
+  python3 ./s3_utilities.py "${BFD_SYNTHEA_OUTPUT_LOCATION}" "upload_prop" "${BFD_SYNTHEA_AUTO_LOCATION}"
+  deactivate
+}
+
+
+#----------------- GO! ------------------#
+# general fail-safe to perform cleanup of any directories and files germane to executing
+# this shell script. Commented out for now
+# clean_up
```

Comment:
Should this be uncommented for the final release?

Reply:
fixed.

Reply:
so I actually added an optional param (defaults to true) to control cleanup; helps a lot if/when the script hits a snag...allows you to comment out some of the steps and re-run.